src/main/java/teammates/ui/controller/StudentCommentsPageAction.java

```diff
@@ -153,7 +158,49 @@ private void setNextPageLink(List<CourseAttributes> courses, int currentIndex){
                 }
             }
         }
-        return feedbackResultBundles;
+        return sortFeedbackBundlesByLastCommentDate(feedbackResultBundles);
+    }
+    
+    /**
+     * @param feedbackResultBundles     The map of FeedbackSessionResultsBundle which is to be sorted 
+     * @return                          Returns a sorted map(LinkedHashMap) of FeedbackSessionResultsBundle,
+     *                                  where the sessions with the latest comment come first.
+     */
+    private Map<String, FeedbackSessionResultsBundle> sortFeedbackBundlesByLastCommentDate(
+            Map<String, FeedbackSessionResultsBundle> feedbackResultBundles) {
+        
+        //this maps all keys of the original map to their corresponding last comment date
+        //this is used during sorting
+        final HashMap<String, Date> lastCommentDateMap = new HashMap<String, Date>();
+        for (Entry<String, FeedbackSessionResultsBundle> entry : feedbackResultBundles
+                .entrySet()) {
+            lastCommentDateMap.put(entry.getKey(), entry.getValue()
+                    .getLastCommentDate());
+        }
+        
+        List<String> list = new ArrayList<String>(
+                feedbackResultBundles.keySet());
+        Collections.sort(list, new Comparator<String>() {
+
+            @Override
+            public int compare(String o1, String o2) {
+                Date date1 = lastCommentDateMap.get(o1);
+                Date date2 = lastCommentDateMap.get(o2);
+                if(date1 == null) {
+                    return 1;
+                }
+                if(date2 == null) {
+                    return -1;
+                }
+                return date2.compareTo(date1);
```

Comment:
Oh ok. I seem to have misinterpreted. I will close this PR and submit another one. Also, since comments for only published sessions are shown, so resultsVisibleFromTime is also a good indicator for the "recentness" of a session. The session which was most recently published, on top. 
Or should I use the session endTime(The session which has the latest endTime on top)?